stl/inc/vector

```diff
@@ -1307,10 +1313,21 @@ public:
     template <class _Iter, enable_if_t<_Is_iterator_v<_Iter>, int> = 0>
     _CONSTEXPR20 void assign(_Iter _First, _Iter _Last) {
         _Adl_verify_range(_First, _Last);
+
+        if (_First == _Last) {
+            clear();

```

Comment:
Clear essentially consists of 4 lines of code. If you would change `_Assign_range` you could inline `clear` and improve debug performance

Reply:
I am not sure about this one though

Reply:
The Standard requires the `assign` family to invalidate all iterators, including the past-the-end iterator, so they can't call `clear()` with its new optimization. I undid the logic changes here.